server/app.mjs

```diff
@@ -532,3 +454,85 @@ function gptscriptConfigPath() {
 
   return path.join(configDir, 'gptscript', 'config.json');
 }
+
+function pickToolName(toolName, existing) {
+  if (!toolName) {
+    toolName = 'external';
+  }
+
+  let testName = toolNormalizer(toolName);
+  while (existing.has(testName)) {
+    testName += '0';
+  }
+  existing.add(testName);
+  return testName;
+}
+
+function toolNormalizer(tool) {
+  const invalidChars = /[^a-zA-Z0-9_]+/g;
+  const validToolName = /^[a-zA-Z0-9]{1,64}$/;
+
+  let parts = tool.split('/');
+  tool = parts[parts.length - 1];
+  if (tool.endsWith('.gpt')) {
+    tool = tool.slice(0, -4);
+  }
+  tool = tool.replace(/^sys\./, '');
+
+  if (validToolName.test(tool)) {
+    return tool;
+  }
+
+  if (tool.length > 55) {
+    tool = tool.slice(0, 55);
+  }
```

Comment:
Just curious, why are you capping it at 55 here instead of 64?

Reply:
To match the logic in gptscript: https://github.com/gptscript-ai/gptscript/blob/1e08c1c2149112ac06d6605d471477955b45a3d0/pkg/types/toolname.go#L38